### ISSUE TITLE
C# expression evaluation in VS fails under very specific circumstances

### ISSUE THREAD

**Issue by SomeAnon42** [dotnet/roslyn#14050]
![bug1](https://cloud.githubusercontent.com/assets/6377116/18807100/9eb80d12-8247-11e6-9d5b-99ff2cd5ee26.png)
![bug2](https://cloud.githubusercontent.com/assets/6377116/18807094/88702972-8247-11e6-80f2-34a0a3eac68e.PNG)

To reproduce this, you're going to need two projects.  The first one is a NETStandard-based library. The second one is a desktop app, either console or WPF. 
- Define two classes in the library as follows:

```
namespace SampleNETStandardLibrary
{
    public class Foo
    {
        public Foo()
        {
            SomeValue = 42;
            Bar = new Bar();
        }

        // We need a property of type that is also defined in this NETStandard-based library.
        // This is very important.
        public Bar Bar { get; }

        // We also need a second property (of any type) in order for the evaluator to fail.
        public int SomeValue { get; }
    }

    public class Bar
    {
        public Bar()
        {
            SomeOtherValue = 69;
        }

        public int SomeOtherValue { get; }
    }
}
```
- That's it for the library. Reference it from your desktop project. You may also need to add project.json.
- Write some code that uses the types defined in the library
  `var instance = new Foo();`

Press F5 and hover over this instance of Foo. There should be a noticeable delay ('Getting DataTip text'). Then you'll see a bunch of question marks instead of the members of the class. 
This is not the case if you use the legacy expression evaluator.

**Sample project**
[ExpressionEvaluatorBug.zip](https://github.com/dotnet/roslyn/files/491040/ExpressionEvaluatorBug.zip)

**Comment by SomeAnon42** [comment]
Just installed VS "15" Preview 5. Wasn't able to reproduce the issue, so seems like it's been fixed in Roslyn 2.0.

**Comment by SomeAnon42** [comment]
project.json is kinda dead, so guess I'll just close this one.
